usefulUtilities/spawnPointZone/entityResources/entityClientScripts/spawnPointZone.js

```diff
@@ -0,0 +1,79 @@
+//
+// spawnPointZone.js
+// 
+// Created by Rebecca Stankus on 05/21/2019
+// Copyright High Fidelity 2019
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+
+(function () {
+    var _this;
+
+    var DEBUG = 1;
+    var HALF = 0.5;
+
+    var SpawnPointZone = function() {
+        _this = this;
+    };
+
+    SpawnPointZone.prototype = {
+        /* When the user loads this zone, they will be moved to random position within a cube specified by its 
+            position and dimensions in the userData of this zone.*/
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            var userData = Entities.getEntityProperties(_this.entityID, 'userData').userData;
+            if (!userData || !JSON.parse(userData).spawnArea.dimensions || !JSON.parse(userData).spawnArea.position) {
```

Comment:
please put `JSON.parse(userData)` around a `try/catch` block, and only parse once.